src/session-managers/SecureSessionManager.ts

```diff
@@ -35,53 +38,45 @@ export class SecureSessionManager {
     }
   }
 
-  async logIn(request: FastifyRequest, user: any, options?: AuthenticateOptions) {
+  async logIn(request: Request, user: any, options?: AuthenticateOptions) {
     const object = await this.serializeUser(user, request)
 
     // Handle @fastify/session to prevent token/CSRF fixation
-    // @ts-expect-error no idea why this does not get typed correctly
     if (request.session.regenerate) {
       if (this.clearSessionOnLogin && object) {
         const keepSessionInfoKeys: string[] = [...this.clearSessionIgnoreFields]
         if (options?.keepSessionInfo) {
           keepSessionInfoKeys.push(...Object.keys(request.session))
         }
-        // @ts-expect-error no idea why this does not get typed correctly
         await request.session.regenerate(keepSessionInfoKeys)
       } else {
-        // @ts-expect-error no idea why this does not get typed correctly
         await request.session.regenerate()
       }
     }
     // Handle @fastify/secure-session against CSRF fixation
     // TODO: This is quite hacky. The best option would be having a regenerate method
     // on secure-session as well
     else if (this.clearSessionOnLogin && object) {
-      const currentFields = request.session.data() || {}
+      const currentData = 'data' in request.session ? request.session.data() : undefined
+      const currentFields = currentData ?? {}
```

Comment:
I think if explicitly set `SessionData` type then the lint issue will go away
```suggestion
      const currentData: SessionData = request.session?.data() ?? {}
```

Reply:
Yeah, that does help 👍